# Step 2: Statistical Analysis of Cleaned Customer Shopping Data

## Introduction

This step focuses on performing a statistical analysis of the cleaned customer shopping dataset. The goal is to understand the central tendency and variability of key numerical variables, as well as the most common values for categorical variables. These insights will help identify patterns in customer demographics, purchasing behavior, and pricing.

### Objectives

1. **Central Tendency**:
   - Calculate the mean, median, and mode for numerical variables (`age`, `quantity`, and `price`).
   - Identify the most common categories (`gender` and `category`).

2. **Variability**:
   - Calculate variance and standard deviation for numerical variables to understand data dispersion.

3. **Interpretation**:
   - Analyze the results to highlight key trends and distributions in the dataset.

---

In [2]:
import pandas as pd
import numpy as np
df = pd.read_csv(r"C:\Users\Zana\Desktop\portfolio_projects\project_1\cleaned_customer_shopping_data.csv")
print(df.head())

  invoice_no customer_id  gender  age  category  quantity    price  \
0    I138884     C241288  Female   28  Clothing         5  1500.40   
1    I317333     C111565    Male   21     Shoes         3  1800.51   
2    I127801     C266599    Male   20  Clothing         1   300.08   
3    I337046     C189076  Female   53     Books         4    60.60   
4    I227836     C657758  Female   28  Clothing         5  1500.40   

  payment_method invoice_date   shopping_mall  
0    Credit Card   2022-08-05          Kanyon  
1     Debit Card   2021-12-12  Forum Istanbul  
2           Cash   2021-11-09       Metrocity  
3           Cash   2021-10-24          Kanyon  
4    Credit Card   2022-05-24  Forum Istanbul  


In [3]:
# Calculate the mean for numerical columns
mean_age = df['age'].mean()
mean_quantity = df['quantity'].mean()
mean_price = df['price'].mean()

mean_age, mean_quantity, mean_price

(43.430813380915566, 2.929177300308155, 528.1295189181749)

In [4]:
# Calculate the median for numerical columns
median_age = df['age'].median()
median_quantity = df['quantity'].median()
median_price = df['price'].median()

median_age, median_quantity, median_price

(43.0, 3.0, 203.3)

In [5]:
# Calculate the mode for categorical and numerical columns
mode_gender = df['gender'].mode()[0]
mode_category = df['category'].mode()[0]
mode_price = df['price'].mode()[0]

mode_gender, mode_category, mode_price

('Female', 'Clothing', 600.16)

In [6]:
# Calculate the variance for numerical columns
variance_age = df['age'].var()
variance_quantity = df['quantity'].var()
variance_price = df['price'].var()

variance_age, variance_quantity, variance_price

(224.89135271120188, 1.959593546514539, 377248.2948068393)

In [7]:
# Calculate the standard deviation for numerical columns
std_age = df['age'].std()
std_quantity = df['quantity'].std()
std_price = df['price'].std()

std_age, std_quantity, std_price

(14.996377986407314, 1.399854830514414, 614.2054174352741)

## Key Results

### **Summary Statistics**

#### 1. Central Tendency:
- **Mean**:
  - Age: **43.43**
  - Quantity: **2.93**
  - Price: **528.13**
- **Median**:
  - Age: **43.0**
  - Quantity: **3.0**
  - Price: **203.30**
- **Mode**:
  - Gender: **Female**
  - Category: **Clothing**
  - Price: **600.16**

#### 2. Variability:
- **Variance**:
  - Age: **224.89**
  - Quantity: **1.95**
  - Price: **377,248.29**
- **Standard Deviation**:
  - Age: **14.99**
  - Quantity: **1.40**
  - Price: **614.20**

---

### **Interpretation**

1. **Age**:
   - The mean and median age are both around 43, suggesting a balanced distribution. 
   - The standard deviation of **14.99** indicates moderate variability in customer age.

2. **Quantity**:
   - The mean and median are close to 3, showing consistency in the number of items purchased.
   - A standard deviation of **1.40** reflects limited spread in purchase quantities.

3. **Price**:
   - A significant difference between the mean (**528.13**) and median (**203.30**) highlights skewness, likely caused by high-value purchases.
   - The high variance (**377,248.29**) and standard deviation (**614.20**) indicate wide variability in item prices.

4. **Categorical Insights**:
   - The most common customer gender is **Female**.
   - The most frequent shopping category is **Clothing**.

---

This analysis provides a foundational understanding of customer demographics and purchasing behavior, setting the stage for more advanced exploratory and predictive analytics.